File: 2018summer/07.07/preliminary/solutions/F/alex.java
// Solution to Runway planning
// Author: Pieter Bootsma, converted to Java by Alex ten Brink

// Time complexity: O(1)
// Memory: O(1)

// @EXPECTED_RESULTS@: CORRECT

// Solution method: calculate it

import java.util.Scanner;

public class alex
{
	public static void main(String[] args)
	{
		Scanner in = new Scanner(System.in);
		int runs = in.nextInt();
		for (int i = 0; i < runs; i++)
		{
			int heading = in.nextInt();

			// Divide by 10, rounding as appropiate
			int runway = (heading + 5) / 10;

			// Limit to range [0,17]
			runway %= 18;

			// Limit to range [1,18]
			if (runway == 0)
			{
				runway = 18;
			}

			// Print leading 0 if less than 10
			if (runway < 10)
			{
				System.out.print(0);
			}

			// Print runway number
			System.out.println(runway);
		}
	}
}

File: 2018summer/07.07/preliminary/solutions/A/alex.cpp
// Solution to Icecream
// Author: Alex ten Brink

// Time complexity: O(log^2(n))
// Memory: O(1)

// @EXPECTED_RESULTS@: CORRECT

// Solution method: trying all options

#include "math.h"
#include <algorithm>
#include <iostream>
using namespace std;

long powermod(long k, long i, long n)
{
	long long acc = 1 % n;
	for (int j = 0; j < i; j++)
	{
		acc =  (acc *k) % n;
	}
	return acc;
}

int main()
{
	int runs;
	cin >> runs;
	while (runs > 0)
	{
		long n, k; bool found = false;
		cin >> n >> k;
		
		int i = 0;
		for (long j = 1; j <= n; j *= 2)
		{
			if (powermod(k, i, n) == 0)
			{
				cout << i << endl;
				found = true;
				break;
			}
			i++;
		}
		if (!found) cout << "unbounded" << endl;

		runs--;
	}
	
	return 0;
}
